app/views/projects/samples/attachments/_attachment.html.erb

```diff
@@ -2,18 +2,82 @@
   id="<%= dom_id(attachment) %>"
   class="bg-white border-b dark:bg-slate-800 dark:border-slate-700"
 >
-  <th scope="row" class="px-6 py-4"><%= link_to attachment.file.filename,
-    download_namespace_project_sample_attachment_path(
-      sample_id: @sample.id,
-      id: attachment.id
-    ),
-    data: {
-      turbo: false
-    } %></th>
-  <td class="px-6 py-4"><%= number_to_human_size(attachment.file.blob.byte_size) %></td>
+  <% if attachment.associated_attachment && attachment.metadata['direction'] == 'forward' %>
+    <td class="px-6 py-4">
+      <div>
+        <div class="flex mb-4 items-center">
+          <%= viral_icon(name: :arrow_right, color: :subdued, classes: "h-6 w-6 ml-0 mr-2") %>
+          <span>
+            <%= link_to attachment.file.filename,
+            download_namespace_project_sample_attachment_path(
+              sample_id: @sample.id,
+              id: attachment.id
+            ),
+            data: {
+              turbo: false
+            } %>
+          </span>
+        </div>
+        <div class="flex items-center">
+          <%= viral_icon(name: :arrow_left, color: :subdued, classes: "h-6 w-6 ml-0 mr-2") %>
+          <span>
+            <%= link_to attachment.associated_attachment_filename,
```

Comment:
I don't think we need a special method for this. You should be able to just call `attachment.associated_attachment.file.filename`. Then we can also remove the `associated_attachment_filename` method in `attachment.rb`

Reply:
Fixed in cb2b6029026c29eec18cc930df8af9cdd894feb7